src/azure-cli-core/azure/cli/core/_help.py

```diff
@@ -80,13 +83,77 @@ def _print_extensions_msg(help_file):
             if help_file.command_source.preview:
                 logger.warning(help_file.command_source.get_preview_warn_msg())
 
-    def print_detailed_help(self, cli_name, help_file):
+    def _print_detailed_help(self, cli_name, help_file):
         AzCliHelp._print_extensions_msg(help_file)
-        self._print_detailed_help(cli_name, help_file)
+        super(AzCliHelp, self)._print_detailed_help(cli_name, help_file)
```

Comment:
Do we actually use this anywhere?

Reply:
It is meant to be used. This method should override Knack's _print_detailed_help method so that extension warning msgs can be printed. See #7606